### ISSUE TITLE
plugin doesn't honor "never indent comments" options

### ISSUE THREAD

**Issue by tominf** [revelc/formatter-maven-plugin#288]
The plugin seems to suffer from the following bug:

https://bugs.eclipse.org/bugs/show_bug.cgi?id=415630

Please, check the link for a more detailed discussion and a proposed solution.

-----

The following description is partially taken from there:

The "Never indent line comments on first column" and "Never indent block comments on first column" work great when used interactively from the Eclipse IDE, but they don't work when invoked via "mvn formatter:format".

Steps to Reproduce:
1) In Eclipse (2018-09 (4.9.0)) go to Window - Preferences.
2) Navigate to Java/Code Style/Formatter
3) Create a new profile (or edit your current one) such it has the following options checked
   [Comments] Never indent line comments on first column
   [Comments] Never indent block comments on first column

4) Create a file named BadStyle.java that looks like this:
```
public class BadStyle {

/* This is a block comment on the first column */
// This is a line comment on the first column

    void DoNothing() {
    }    
}
```

5) Confirm that if you format the file with Ctrl+Shift+F, the comments don't move.

6) Export the current formatter settings

7) Configure the plugin to use the exported settings:
```
<plugin>
    <groupId>net.revelc.code.formatter</groupId>
    <artifactId>formatter-maven-plugin</artifactId>
    <version>2.8.0</version>
    ...
    <configuration>
        <configFile>.../exported_config.xml</configFile>
    </configuration>
</plugin>
```

8) Format the file via "mvn formatter:format"

What Happens:
The comments in BadStyle.java were indented.

Expected Result:
The comments in BadStyle.java remain on the first column.

-----

The above mentioned bug discussion also proposes a solution that seems to have worked for command line Eclipse. Since this plugin is always (?) used for formatting existing code rather than newly generated one, this solution should work here as well. In that case, it would be necessary to replace:

`this.formatter = ToolFactory.createCodeFormatter(options);`

with

`this.formatter = ToolFactory.createCodeFormatter(options, ToolFactory.M_FORMAT_EXISTING);`

in /src/main/java/net/revelc/code/formatter/java/JavaFormatter.java @ line 39.

WARNING: I have no experience with this plugin's code though!!!

**Comment by ctubbsii** [comment]
Hi @tominf ! Thank you so much for the bug report. Do you have a small maven project on GitHub (or can you create one) that demonstrates the problem, that uses the current formatter that I can clone in order to verify the bug and the proposed fix? It would be greatly appreciated.

**Comment by ctubbsii** [comment]
@tominf If you can provide a sample maven project, I can test PR#290 using that project. Otherwise, I'll have to create my own project to test it with, and I might not have time for a few days.

**Comment by tominf** [comment]
@ctubbsii Thanks for your reply! I've created the small GitHub project:

https://github.com/tominf/test_format

If you "mvn formatter:format", the first two comments in BadStyle.java should stay where they are (in 2.8.0 they get indented). The other two should get indented, which is ok.

**Comment by ctubbsii** [comment]
@tominf Thanks! I tested against that project, and was able to verify the problem, and that the proposed fix does indeed solve the problem. Expect this to be fixed in 2.8.1.

**Comment by tominf** [comment]
Great! Thanks a lot!
